Src/FluentAssertions/Execution/XUnit2TestFramework.cs

```diff
@@ -1,40 +0,0 @@
-﻿using System;
-using System.Diagnostics.CodeAnalysis;
-using System.Reflection;
-
-namespace FluentAssertions.Execution;
-
-/// <summary>
-/// Implements the XUnit (version 2) test framework adapter.
-/// </summary>
-internal class XUnit2TestFramework : ITestFramework
-{
-    private Assembly assembly;
-
-    public bool IsAvailable
-    {
-        get
-        {
-            try
-            {
-                // For netfx the assembly is not in AppDomain by default, so we can't just scan AppDomain.CurrentDomain
-                assembly = Assembly.Load(new AssemblyName("xunit.assert"));
```

Comment:
We added this in #384 to correctly throw `XUnitException` despite `xunit.assert.dll` not yet being loaded into the App Domain for netfx.

The reason this PR works is due to how https://github.com/fluentassertions/fluentassertions/pull/2712/files#diff-69d48f2874ce90a847fb038d9eb42b104dba380c129f7babfde78e9a3d73e65aL16-L17 changed the test. 
By explicitly referencing `XunitException`, we now trigger loading the `xunit.assert` dll.

We should revert those simplifications.
At least for xunit as that's the only framework that is spread across multiple dll's, but I'd say we do it for all test frameworks.

Reply:
> We should revert those simplifications.

Indeed, I figured out why `XunitException` was not referenced while working on this pull request. I'll revert #2712 as part of this pull request and add a comment _why_ `act.Should().Throw<XunitException>();` should not be used.